### ISSUE TITLE
How to get blur on menus ?

### ISSUE THREAD

**Issue by aislanmaia** [tsujan/Kvantum#215]
Right now, with KDE 5.13 release the Breeze theme engine can blurring menus. But trying to get the same effect with Kvantum themes seems to be not possible, just windows background can get blur.

Is possible to achieve the same with Kvantum, or this is some effect not available right now?

I'm already have enabled all the blur and translucent related settings in Kvantum manager but no blurring to menus (right click menus, dropdown menus, etc).

**Comment by tsujan** [comment]
Menu/tooltip blurring has been supported by Kvantum under KDE for a very long time:

![menu_blurring](https://user-images.githubusercontent.com/981076/41458436-d7f0cdb4-709b-11e8-9e9b-795d156fac2a.png)

The theme should have a translucent menu (as in the above screenshot) and this option should be checked:  `Kvantum Manager → Configure Active Theme → Compositing & General Look → Blurring for menus and tooltips`.

Of course, KDE's blurring effect should be enabled. You said that window bg is blurred. So, KDE's blur effect is enabled in your case. I guess the theme you use doesn't have translucent menus; otherwise, its menu bg should be blurred automatically because window blurring enables menu/tooltip blurring.

Unlike window bg, the opacity of menu bg can't be changed by Kvantum manager.

**Comment by tsujan** [comment]
BTW, the Plasma version I have is 5.12.5. I don't think blurring can be changed in that way with 5.13 (a bug?) but will tell you if that's the case.

**Comment by aislanmaia** [comment]
@tsujan thanks for the reply. Yeah, blur in general is working definitely for me (effect activated already). I've seen this options you have said above, but `Blurring for menus and tooltips` option doesn't work for some themes I've tested out. One for instance, is the Adapta (and others) theme. Anyways, this could be just a issue theme, after all.

**Comment by varlesh** [comment]
@tsujan I confirm this, if enabled blur effect on KDE - MenuItem and Tooltip disabled transparency  on Kvantum.
For example, kvantum theme KvFlat
Disabled Blur on KDE:
![image](https://user-images.githubusercontent.com/8083855/41805496-9a255f6e-76bc-11e8-82b2-597237a081ea.png)
Enabled Blur on KDE:
![image](https://user-images.githubusercontent.com/8083855/41805520-05e4f9b2-76bd-11e8-97fb-604e6ef3df65.png)

**Comment by varlesh** [comment]
Plasma 5.13.1
Kvantum 0.10.6git20180503ubuntu16.04

**Comment by tsujan** [comment]
@varlesh Actually, those menus ARE translucent and blurred but KDE's blurring code is changed in a bad way. This screenshot is taken with KvFlat too but I've made KDE blurring much milder:

![blurring](https://user-images.githubusercontent.com/981076/41805678-2605ddfe-76c4-11e8-8661-ab6095167835.png)

As you see, the menu is blurred but too strongly.

KDE's blurring has been reliable since KDE4 but, with Plasma 5.13.1, it isn't so. This isn't a Kvantum issue and happens with QtCurve too. In the following screenshot, the opacity of QtCurve menus is exactly equal to that of its windows and is 80% but menus look opaque:

![qtcurve](https://user-images.githubusercontent.com/981076/41805839-01268518-76c6-11e8-9a3a-1f285ea1e140.png)

In short, this is a KDE regression.

**Comment by tsujan** [comment]
It's a shame that KDE has so many problems.... That's why it isn't my main DE anymore.

**Comment by varlesh** [comment]
Thx for info.

**Comment by tsujan** [comment]
You're welcome.

It's sad that KDE blurring is damaged :( It was one of the most stable things about KDE.

**Comment by varlesh** [comment]
It's not all. On aurorae decoration with enabled blur bugged shadows and angles.
![image](https://user-images.githubusercontent.com/8083855/41806578-a6d6f316-76d1-11e8-9086-266a4a059604.png)
![image](https://user-images.githubusercontent.com/8083855/41806576-9c12a51a-76d1-11e8-8a4e-ae1172777770.png)

**Comment by tsujan** [comment]
It should be reported to the KDE bug tracker.

**Comment by tsujan** [comment]
@varlesh Is it reported to KDE? I can't do that because KDE isn't my main DE anymore. The sooner they know about these regressions, the better.

**Comment by tsujan** [comment]
Another KDE issue: https://github.com/tsujan/Kvantum/issues/218

**Comment by aislanmaia** [comment]
Thanks for the info @tsujan . I don't know how to report this appropriately to KDE.

**Comment by tsujan** [comment]
> because if I use the Breeze engine + Breeze theme the blur works (and all menus) fine.

But Breeze's menus aren't translucent to be blurred. A screenshot?

**Comment by varlesh** [comment]
@aislanmaia all worked. Need reduce blur and noise on effect.
![image](https://user-images.githubusercontent.com/8083855/42113878-0125f066-7bfe-11e8-8602-2bf09db99f00.png)

**Comment by tsujan** [comment]
@varlesh No, there's a regression in KDE blur effect. Reducing the noise and blur is just an ugly workaround -- I'd checked it before.

**Comment by aislanmaia** [comment]
@varlesh this is the original Adaita or some variation theme? Have you reduced in KDE's settings or in Kvantum ?

**Comment by varlesh** [comment]
@aislanmaia It's variration Materia KDE Theme.
I'm add opacity on some widgets in kvantum SVG (menu, tooltip, header) and fix some bugs. 
For better looking menu & tooltip need change blur and noise strength to 4 position on KDE systemsettings blur effect.
![image](https://user-images.githubusercontent.com/8083855/42131970-32c7cd20-7d1f-11e8-9a80-d24f12cebf0a.png)

**Comment by varlesh** [comment]
@aislanmaia If menu or tooltip not have opacity - transparency and blur not applied on Kvantum for it widgets.

**Comment by aislanmaia** [comment]
Great work @varlesh . Thanks for the tips. I wonder if the same can be applied for the original Adaita. I've tested the Materia as well and got the same issue. How can I change the SVG file? I've looked in the Adaita svg file with Inkscape but for me was very confusing the images that appeared there.

**Comment by tsujan** [comment]
OK, I found a workaround for the buggy blur effect of Plasma 5.13 without sacrificing anything :)

I'll add a key (to Kvantum Manager) for reducing menu opacity (`reduce_menu_opacity`). In the following screenshot, I've used the default values of the blur effect and set the value of that key to 30 with `KvCurves3d`:

![new_blur](https://user-images.githubusercontent.com/981076/42294994-46dd0382-7ffb-11e8-8554-454937f4ecbe.png)

In this way, even if the theme doesn't have translucent SVG elements for menu, its menu can be made translucent :)

I'll tell everyone here when the change is made.

It's good that KDE devs pay attention to blurring but it's bad that their new codes are so buggy (new developers?!).

**Comment by tsujan** [comment]
Done in https://github.com/tsujan/Kvantum/commit/75da091f6104703b49e3fa156bbeb9c162a468bf

**Comment by tsujan** [comment]
I found the commit for this new blur effect: https://cgit.kde.org/kwin.git/commit/?id=0179f741bbd4cc01da5866fdf7b6a302869a099b

Apparently, it's much stronger than the previous effect, at least with SVG. The good thing about it is that it should work under wayland too. If so, KDE has the first wayland compositor with a blur effect and that's very cool :)

I might enable Kvantum window translucency under wayland (it's disabled in the code) to see what happens -- the last time, it was terrible ;)
